lib/landrush/cap/host/windows/configure_visibility_on_host.rb

```diff
@@ -19,46 +20,82 @@ def configure_visibility_on_host(env, ip, tlds)
-          #    Register with which suffix:           Primary only
-          #    WINS servers configured through DHCP: None
-          # \n\n
-          # ...
-          def get_network_name(ip)
-            cmd_out = `netsh interface ip show config`
-            network_details = cmd_out.split(/^$/).reject(&:empty?).select do |settings|
-              begin
-                lines = settings.split(/\n/).reject(&:empty?)
-                subnet = lines[3]
-                next false unless subnet =~ /Subnet Prefix/
-
-                mask = IPAddr.new(subnet.match(%r{.* (\d{1,3}\.\d{1,3}\.\d{1,3}.\d{1,3}/\d{1,3}).*}).captures[0])
-                address = IPAddr.new(ip)
-
-                mask.include?(address)
-              rescue StandardError
-                false
+          # Given an IP determines the network adapter guid, if any.
+          # We can't use netsh due to its output being locale dependant.
+          # Using Windows registry we achieve the same result of netsh in
+          # whatever language Windows is.
+          def get_network_guid(address)
+            address = IPAddr.new(address.to_s)
+
+            interfaces do |interface_guid|
+              interface = open_interface("#{INTERFACES}\\#{interface_guid}")
+              if_ip, if_mask = interface_address(interface)
+
+              next if if_ip.nil?
+
+              if_net = IPAddr.new("#{if_ip}/#{if_mask}")
+              return interface_guid if if_net.include?(address)
+            end
+          rescue StandardError
+            nil
+          end
+
+          private
+
+          # Fetch interfaces from registry
+          # This is a separate method to make testing easier
+          def interfaces
+            Win32::Registry::HKEY_LOCAL_MACHINE.open(INTERFACES) do |interfaces|
+              interfaces.each_key do |guid, _|
+                yield guid
               end
             end
-            return nil if network_details[0].nil? || network_details[0].empty?
+          end
 
-            network_details[0].split(/\n/).reject(&:empty?)[0].match(/Configuration for interface "(.*)"/).captures[0].strip
+          # Opens an interface key from registry
+          # This is a separate method to make testing easier
+          def open_interface(guid_path)
+            Win32::Registry::HKEY_LOCAL_MACHINE.open(guid_path)
           end
 
-          private
+          # Fetches IP/mask info from registry from a given registry entry
+          # This is a separate method to make testing easier
+          def interface_address(interface)
+            dhcp_enabled = interface.read('EnableDHCP')[1]
+
+            if dhcp_enabled
+              if_ip = interface.read('DhcpIPAddress')[1]
+              if_mask = interface.read('DhcpSubnetMask')[1]
+            else
+              if_ip = interface.read('IPAddress')[1]
+              if_mask = interface.read('SubnetMask')[1]
+            end
+
+            if if_ip.is_a? Array
```

Comment:
What are the elements of the array? Do you know in which cases this occurs?

Reply:
From what I saw, this only occurs when you have DHCP enabled.
The elements are... a single IP.
I didn't find why Windows stores static IPs as a single-value and DHCP as multi-value in registry.

Reply:
Hmm, a bit random, but fair enough I guess

Reply:
After reading around, it looks like this is like this for a reason: if the interface is not using DHCP, you can have multiple IPs on the same interface.
This makes sense to me. But I don't have a use case for this, so I am not able to properly test. I'm pretty sure that this (multiple IPs on the same NIC) will break things, but I think that this is such an edge case that I shouldn't care about it *right now*. :)